app/models/galleries_icon.rb

```diff
@@ -2,4 +2,12 @@ class GalleriesIcon < ActiveRecord::Base
   belongs_to :icon
   belongs_to :gallery
   accepts_nested_attributes_for :icon, allow_destroy: true
+
+  after_destroy :unset_has_gallery
+
+  def unset_has_gallery
+    icon = Icon.find_by_id(icon_id)
+    return if icon.galleries.present?
+    icon.update_attributes(has_gallery: false)
+  end
```

Comment:
Using `gallery#edit` to destroy a `GalleriesIcon` wasn't unsetting `has_gallery`… I'm not sure if this is a patch that should be applied to the Rails 3 thing; I'll investigate